// three/src/constants.ts
interface constantsType {
  [key: string]: any,
}

const constants: constantsType = {
  REVISION : '143',
  MOUSE : { LEFT: 0, MIDDLE: 1, RIGHT: 2, ROTATE: 0, DOLLY: 1, PAN: 2 },
  TOUCH : { ROTATE: 0, PAN: 1, DOLLY_PAN: 2, DOLLY_ROTATE: 3 },
  CullFaceNone : 0,
  CullFaceBack : 1,
  CullFaceFront : 2,
  CullFaceFrontBack : 3,
  BasicShadowMap : 0,
  PCFShadowMap : 1,
  PCFSoftShadowMap : 2,
  VSMShadowMap : 3,
  FrontSide : 0,
  BackSide : 1,
  DoubleSide : 2,
  FlatShading : 1,
  SmoothShading : 2,
  NoBlending : 0,
  NormalBlending : 1,
  AdditiveBlending : 2,
  SubtractiveBlending : 3,
  MultiplyBlending : 4,
  CustomBlending : 5,
  AddEquation : 100,
  SubtractEquation : 101,
  ReverseSubtractEquation : 102,
  MinEquation : 103,
  MaxEquation : 104,
  ZeroFactor : 200,
  OneFactor : 201,
  SrcColorFactor : 202,
  OneMinusSrcColorFactor : 203,
  SrcAlphaFactor : 204,
  OneMinusSrcAlphaFactor : 205,
  DstAlphaFactor : 206,
  OneMinusDstAlphaFactor : 207,
  DstColorFactor : 208,
  OneMinusDstColorFactor : 209,
  SrcAlphaSaturateFactor : 210,
  NeverDepth : 0,
  AlwaysDepth : 1,
  LessDepth : 2,
  LessEqualDepth : 3,
  EqualDepth : 4,
  GreaterEqualDepth : 5,
  GreaterDepth : 6,
  NotEqualDepth : 7,
  MultiplyOperation : 0,
  MixOperation : 1,
  AddOperation : 2,
  NoToneMapping : 0,
  LinearToneMapping : 1,
  ReinhardToneMapping : 2,
  CineonToneMapping : 3,
  ACESFilmicToneMapping : 4,
  CustomToneMapping : 5,
  
  UVMapping : 300,
  CubeReflectionMapping : 301,
  CubeRefractionMapping : 302,
  EquirectangularReflectionMapping : 303,
  EquirectangularRefractionMapping : 304,
  CubeUVReflectionMapping : 306,
  RepeatWrapping : 1000,
  ClampToEdgeWrapping : 1001,
  MirroredRepeatWrapping : 1002,
  NearestFilter : 1003,
  NearestMipmapNearestFilter : 1004,
  NearestMipMapNearestFilter : 1004,
  NearestMipmapLinearFilter : 1005,
  NearestMipMapLinearFilter : 1005,
  LinearFilter : 1006,
  LinearMipmapNearestFilter : 1007,
  LinearMipMapNearestFilter : 1007,
  LinearMipmapLinearFilter : 1008,
  LinearMipMapLinearFilter : 1008,
  UnsignedByteType : 1009,
  ByteType : 1010,
  ShortType : 1011,
  UnsignedShortType : 1012,
  IntType : 1013,
  UnsignedIntType : 1014,
  FloatType : 1015,
  HalfFloatType : 1016,
  UnsignedShort4444Type : 1017,
  UnsignedShort5551Type : 1018,
  UnsignedInt248Type : 1020,
  AlphaFormat : 1021,
  RGBFormat : 1022,
  RGBAFormat : 1023,
  LuminanceFormat : 1024,
  LuminanceAlphaFormat : 1025,
  DepthFormat : 1026,
  DepthStencilFormat : 1027,
  RedFormat : 1028,
  RedIntegerFormat : 1029,
  RGFormat : 1030,
  RGIntegerFormat : 1031,
  RGBAIntegerFormat : 1033,
  
  RGB_S3TC_DXT1_Format : 33776,
  RGBA_S3TC_DXT1_Format : 33777,
  RGBA_S3TC_DXT3_Format : 33778,
  RGBA_S3TC_DXT5_Format : 33779,
  RGB_PVRTC_4BPPV1_Format : 35840,
  RGB_PVRTC_2BPPV1_Format : 35841,
  RGBA_PVRTC_4BPPV1_Format : 35842,
  RGBA_PVRTC_2BPPV1_Format : 35843,
  RGB_ETC1_Format : 36196,
  RGB_ETC2_Format : 37492,
  RGBA_ETC2_EAC_Format : 37496,
  RGBA_ASTC_4x4_Format : 37808,
  RGBA_ASTC_5x4_Format : 37809,
  RGBA_ASTC_5x5_Format : 37810,
  RGBA_ASTC_6x5_Format : 37811,
  RGBA_ASTC_6x6_Format : 37812,
  RGBA_ASTC_8x5_Format : 37813,
  RGBA_ASTC_8x6_Format : 37814,
  RGBA_ASTC_8x8_Format : 37815,
  RGBA_ASTC_10x5_Format : 37816,
  RGBA_ASTC_10x6_Format : 37817,
  RGBA_ASTC_10x8_Format : 37818,
  RGBA_ASTC_10x10_Format : 37819,
  RGBA_ASTC_12x10_Format : 37820,
  RGBA_ASTC_12x12_Format : 37821,
  RGBA_BPTC_Format : 36492,
  LoopOnce : 2200,
  LoopRepeat : 2201,
  LoopPingPong : 2202,
  InterpolateDiscrete : 2300,
  InterpolateLinear : 2301,
  InterpolateSmooth : 2302,
  ZeroCurvatureEnding : 2400,
  ZeroSlopeEnding : 2401,
  WrapAroundEnding : 2402,
  NormalAnimationBlendMode : 2500,
  AdditiveAnimationBlendMode : 2501,
  TrianglesDrawMode : 0,
  TriangleStripDrawMode : 1,
  TriangleFanDrawMode : 2,
  LinearEncoding : 3000,
  sRGBEncoding : 3001,
  BasicDepthPacking : 3200,
  RGBADepthPacking : 3201,
  TangentSpaceNormalMap : 0,
  ObjectSpaceNormalMap : 1,
  
  // Color space string identifiers, matching CSS Color Module Level 4 and WebGPU names where available.
  NoColorSpace : '',
  SRGBColorSpace : 'srgb',
  LinearSRGBColorSpace : 'srgb-linear',
  
  ZeroStencilOp : 0,
  KeepStencilOp : 7680,
  ReplaceStencilOp : 7681,
  IncrementStencilOp : 7682,
  DecrementStencilOp : 7683,
  IncrementWrapStencilOp : 34055,
  DecrementWrapStencilOp : 34056,
  InvertStencilOp : 5386,
  
  NeverStencilFunc : 512,
  LessStencilFunc : 513,
  EqualStencilFunc : 514,
  LessEqualStencilFunc : 515,
  GreaterStencilFunc : 516,
  NotEqualStencilFunc : 517,
  GreaterEqualStencilFunc : 518,
  AlwaysStencilFunc : 519,
  
  StaticDrawUsage : 35044,
  DynamicDrawUsage : 35048,
  StreamDrawUsage : 35040,
  StaticReadUsage : 35045,
  DynamicReadUsage : 35049,
  StreamReadUsage : 35041,
  StaticCopyUsage : 35046,
  DynamicCopyUsage : 35050,
  StreamCopyUsage : 35042,
  
  GLSL1 : '100',
  GLSL3 : '300 es',
  
  _SRGBAFormat : 1035, // fallback for WebGL 1
  
}

export default constants;
